src/AspNetCoreModuleV2/AspNetCore/HandlerResolver.cpp

```diff
@@ -225,26 +236,26 @@ HandlerResolver::FindNativeAssemblyFromHostfxr(
         }
     }
 
-    RETURN_IF_FAILED(struNativeSearchPaths.SyncWithBuffer());
+    struNativeSearchPaths.resize(struNativeSearchPaths.find(L'\0'));
 
     fFound = FALSE;
 
     // The native search directories are semicolon delimited.
     // Split on semicolons, append aspnetcorerh.dll, and check if the file exists.
-    while ((intIndex = struNativeSearchPaths.IndexOf(L";", intPrevIndex)) != -1)
+    while ((intIndex = struNativeSearchPaths.find(L';', intPrevIndex)) != std::wstring::npos)
     {
-        RETURN_IF_FAILED(struNativeDllLocation.Copy(&struNativeSearchPaths.QueryStr()[intPrevIndex], intIndex - intPrevIndex));
+        auto path = struNativeSearchPaths.substr(intPrevIndex, intIndex - intPrevIndex);
 
-        if (!struNativeDllLocation.EndsWith(L"\\"))
+        if (!path.empty() && !(path[path.length()] == L'\\'))
```

Comment:
Shouldn't this be path.length() -1? Consider using std::filesystem::path here too.

Reply:
Yep